float64 position
string direction
uint32 period
bool enabled